components/analytics-io/org.wso2.carbon.analytics.restapi/src/main/java/org/wso2/carbon/analytics/restapi/resources/AnalyticsResource.java

```diff
@@ -1177,6 +1176,39 @@ public Response getTablePersistName(@PathParam("tableName") String tableName,
 		}
 	}
 
+	/**
+	 * Gets the actual table name from the database persist name for a table
+	 *
+	 * @param persistName the persisted table name for which the actual name needs to be inferred
+	 * @return HTTP response containing the actual table name
+	 * @throws AnalyticsException
+	 */
+	@GET
+	@Produces({MediaType.TEXT_PLAIN})
+	@Path("tables/{persistName}/actualName")
+    public Response getTableActualName(@PathParam("persistName") String persistName,
+                                       @HeaderParam(AUTHORIZATION_HEADER) String authHeader)
+            throws AnalyticsException {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Invoking getTableActualName for table : " + persistName);
+        }
+        String username = authenticate(authHeader);
+		String tenantDomain = MultitenantUtils.getTenantDomain(username);
+		List<String> tables = Utils.getAnalyticsDataAPIs().listTables(username);
+        try {
+            int tenantId = Utils.getRealmService().getTenantManager().getTenantId(tenantDomain);
+            for (String tableName : tables) {
+                String candidate = GenericUtils.generateTableUUID(tenantId, tableName);
+                if (candidate.equals(persistName)) {
+                    return Response.ok(tableName).build();
+                }
+            }
+            return Response.ok().build();
```

Comment:
Don't we need to say why we are not getting table name as payload?

Reply:
Done.